crates/mpt/src/db/mod.rs

```diff
@@ -307,11 +352,27 @@ where
         Ok(int_slot)
     }
 
-    fn block_hash(&mut self, _: U256) -> Result<B256, Self::Error> {
-        // match self.db.block_hashes.entry(number) {
-        //     Entry::Occupied(entry) => Ok(*entry.get()),
-        //     Entry::Vacant(_) => anyhow::bail!("Block hash for number not found"),
-        // }
-        unimplemented!("Block hash not implemented; Need to unroll the starting block hash for this operation.")
+    fn block_hash(&mut self, block_number: U256) -> Result<B256, Self::Error> {
```

Comment:
There's no way to do this without walking back huh?

Reply:
Yep, that's the way to do it in a verifiable way, starting from the latest parent block hash. In a normal client, you could just consult the DB, but with the execution for this DB being stateless, we potentially need to verifiably fetch data prior to the initialization point (i.e., in the first block we execute, there's a `BLOCKHASH <current_block> - 200` opcode.